### ISSUE TITLE
JVM cannot handle https connections

### ISSUE THREAD

**Issue by kekkaishivn** [lsds/sgx-lkl#33]
I am trying to run a simple java application establish https connection and I got this exception: 

java.security.ProviderException: Could not initialize NSS
	at sun.security.pkcs11.SunPKCS11.<init>(SunPKCS11.java:223)
	at sun.security.pkcs11.SunPKCS11.<init>(SunPKCS11.java:103)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at sun.security.jca.ProviderConfig$2.run(ProviderConfig.java:224)
	at sun.security.jca.ProviderConfig$2.run(ProviderConfig.java:206)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.security.jca.ProviderConfig.doLoadProvider(ProviderConfig.java:206)
	at sun.security.jca.ProviderConfig.getProvider(ProviderConfig.java:187)
	at sun.security.jca.ProviderList.loadAll(ProviderList.java:282)
	at sun.security.jca.ProviderList.removeInvalid(ProviderList.java:299)
	at sun.security.jca.Providers.getFullProviderList(Providers.java:173)
	at java.security.Security.getProviders(Security.java:485)
	at HelloWorld.main(HelloWorld.java:44)
Caused by: java.io.IOException: Error loading shared library libnss3.so: No such file or directory
	at sun.security.pkcs11.Secmod.nssLoadLibrary(Native Method)
	at sun.security.pkcs11.Secmod.initialize(Secmod.java:210)
	at sun.security.pkcs11.SunPKCS11.<init>(SunPKCS11.java:218)

Is there anyway I can set a custom library to the jre provided by sgx-lkl.

**Comment by cpriebe** [comment]
Hi @kekkaishivn,

thanks for opening this issue. As the error message implies the disk image is missing the nss library. Solving this is simply a matter of providing the libnss3.so library. I have fixed this for the HelloWorld example as people probably use it as a starting point for running the JVM with SGX-LKL and might run into the same issue as you otherwise.

https://github.com/lsds/sgx-lkl/blob/664eb25ae087809e371763a3b946df2a18681869/apps/jvm/helloworld-java/buildenv.sh#L6

This has been fixed in 664eb25.

Let me know if this (not) helped.
